SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -83,108 +127,115 @@ TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActor
 
 	const WorkerRequirementSet AuthoritativeWorkerRequirementSet = { WorkerAttributeOrSpecificWorker };
 
-	WorkerRequirementSet ReadAcl;
-	if (Class->HasAnySpatialClassFlags(SPATIALCLASS_ServerOnly))
-	{
-		ReadAcl = AnyServerRequirementSet;
-	}
-	else if (Actor->IsA<APlayerController>())
-	{
-		ReadAcl = AnyServerOrOwningClientRequirementSet;
-	}
-	else
-	{
-		ReadAcl = AnyServerOrClientRequirementSet;
-	}
-
-	WriteAclMap ComponentWriteAcl;
-	ComponentWriteAcl.Add(SpatialConstants::POSITION_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
-	ComponentWriteAcl.Add(SpatialConstants::INTEREST_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
-	ComponentWriteAcl.Add(SpatialConstants::SPAWN_DATA_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
-	ComponentWriteAcl.Add(SpatialConstants::DORMANT_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
-	ComponentWriteAcl.Add(SpatialConstants::SERVER_TO_SERVER_COMMAND_ENDPOINT_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
-	ComponentWriteAcl.Add(SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
-	ComponentWriteAcl.Add(SpatialConstants::UNREAL_METADATA_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
-	ComponentWriteAcl.Add(SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
-	ComponentWriteAcl.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID, AnyServerRequirementSet);
-	ComponentWriteAcl.Add(SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
+	EntityAcl* Acl = (static_cast<EntityAcl*>(EntityComps.ModifiableComponents[EntityAcl::ComponentId].Get())); // no idea which cast
+	Acl->ComponentWriteAcl.Add(SpatialConstants::POSITION_COMPONENT_ID, AuthoritativeWorkerRequirementSet);
```

Comment:
You could eliminate a Ton of Deltas by getting a ref on stack to Acl->ComponentWriteACL
(I will have to merge my changes for the routing worker, that is why I care)

Reply:
Ah yeah, I even wanted to do it previously, but forgot. Good point, will try to do

Reply:
Done now